module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include <algorithm>
+#include <iostream>
+
+#include "compression.hpp"
+
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& inputMap){
```

Comment:
You can make some fancy alias for these long, complex type like `using smth = std::vector<std::pair<uint8_t, uint8_t>>`